lib/aruco_detector.hpp

```diff
@@ -5,13 +5,24 @@
 #include <memory>
 #include "classification_model.hpp"
 
+struct ArucoParams {
+    double markerSize;
+    cv::Mat cameraMatrix;
+    cv::Mat distCoeffs;
+    std::vector<int> ids;
+    std::vector<std::vector<cv::Point2f>> corners;
+    std::vector<cv::Vec3d> rvecs;
+    std::vector<cv::Vec3d> tvecs;
+};
+
 class Aruco_Detector {
 public:
     Aruco_Detector(const cv::Mat& cameraMatrix, int dictionaryType, float markerSize = 0.15)
         : cameraMatrix(cameraMatrix), dictionaryType(dictionaryType), markerSize(markerSize) {}
```

Comment:
Since the `ArucoParams` struct already exists, the constructor can just take a object of `ArucoParams` instead of the individual parameters.